WalletWasabi/WabiSabi/Client/Batching/PendingPayment.cs

```diff
@@ -0,0 +1,20 @@
+using System.Collections.Immutable;
+using NBitcoin;
+using WalletWasabi.Extensions;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.WabiSabi.Client.Batching;
+
+// Represents a payment that is waiting to be made. 
+public record PendingPayment(IDestination Destination, Money Amount)
+	: Payment(Destination, Amount)
+{
+	public Guid Id { get; } = Guid.NewGuid();
+	
+	public bool FitParameters(ImmutableSortedSet<ScriptType> allowedOutputTypes, MoneyRange allowedOutputAmounts) =>
+		allowedOutputTypes.Contains(Destination.ScriptPubKey.GetScriptType()) &&
```

Comment:
No. The concept of robustness is wrong. A method has to do what it has to do, and that's only possible if it has valid parameters. The only way an invalid/unknown ScriptPubKey can be here is because the method was called with invalid data, that means that the caller didn't validate the input from the user, or its caller didn't do it, and so on. 

The ScriptPubKey type must be checked/validated much before and much closer to the system bounds. The user must be notified that the script that is invalid so he can fix it.